CoseHandler/Extensions/FileInfoExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace CoseX509;
+
+using System;
+using System.IO;
+using System.Threading;
+using CoseSign1;
+
+public static class FileInfoExtensions
+{
+    /// <summary>
+    /// Loads the content of a file into a byte array after making sure the file exists, is not empty, and is not locked by another process.
+    /// </summary>
+    /// <param name="f">The file to read from.</param>
+    /// <param name="writeTo">The output target to write status messages to. Default is STDOUT.</param>
+    /// <returns>The file content.</returns>
+    public static byte[] GetBytesResilient(this FileInfo f, OutputTarget? writeTo = null) => GetBytesOrStream(f, false, writeTo).Item1!;
+
+    /// <summary>
+    /// Loads the content of a file into a <see cref="FileStream"/> after making sure the file exists, is not empty, and is not locked by another process.
+    /// </summary>
+    /// <param name="f">The file to read.</param>
+    /// <param name="writeTo">The output target to write status messages to. Default is STDOUT.</param>
+    /// <returns>The file content.</returns>
+    public static FileStream? GetStreamResilient(this FileInfo f, OutputTarget? writeTo = null) => GetBytesOrStream(f, true, writeTo).Item2;
+
+    private static (byte[]?, FileStream?) GetBytesOrStream(FileInfo f, bool isStream, OutputTarget? writer)
+    {
+        // Make sure the file exists, allowing retries in case it hasn't hit the disk yet.
+        DateTime startTime = DateTime.Now;
+        while (f is null || !f.Exists)
+        {
+            Thread.Sleep(100);

```

Comment:
should we define this backoff time value in a constant somewhere? I noticed we are using 250 below as well, so might be useful to have a comment or variable to describe why these values are what they are. Might also make sense to allow the user to control this value (within an acceptable range), with the current values as defaults - same for the max wait time passed into `OutOfTime()`

Reply:
Done.